src/server.c

```diff
@@ -3628,9 +3634,9 @@ int processCommand(client *c) {
      * 1) The sender of this command is our master.
      * 2) The command has no key arguments. */
     if (server.cluster_enabled &&
-        !(c->flags & CLIENT_MASTER) &&
+        !mustObeyClient(c) &&
         !(c->flags & CLIENT_SCRIPT &&
-          server.script_caller->flags & CLIENT_MASTER) &&
+          mustObeyClient(server.script_caller)) &&
```

Comment:
IIUC `script_caller` would never reach here, script just uses `call()` directly, I remember we discussed it but forgot the details.

Reply:
it seems this was added here: 2f4240b9d9. i guess this part of that commit is a mistake.
@MeirShpilraien please ack, and let's delete it.

Reply:
Yes, I agree it will never reach here with a script client. We can delete it.